website__layout  (rabiulawal1422)

shepherd1422 opened the pull request:
[UI design(error -Популярные товары скролл)

shepherd1422 committed:
коммит перед началом к входу ошибочным приступам , которые надо решать побыстрее,перед началом следущего созвона

коммит перед началом к входу ошибочным приступам , которые надо решать побыстрее,перед началом следущего созвона

shepherd1422 committed:
сверстал до 3-го блока (.item-cards__grid-right) еше не дошел до блока популярные товары^^

shepherd1422 committed:
перед исправлением див ---item-cards__grid

shepherd1422 committed:
div block is fixed

shepherd1422 committed:
UI design(error -Популярные товары скролл)

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -24,6 +23,12 @@
 }
 .header-nav {
     display: flex;
+    font-family: Inter;
```
Вместо того, что бы в каждом классе прописывать font-family: Inter, мы задаем элементу body это свойство, благодаря чему каждый текстовый элемент уже будет иметь свойство font-family: Inter

rabiulawal1422 replied:
<img width="1283" alt="Screenshot 2024-08-18 at 12 47 55 PM" src="https://github.com/user-attachments/assets/1ad1b5ff-357b-46d8-b749-35ef8c884dab">

shepherd1422 replied:
я сделал)
<img width="837" alt="Screenshot 2024-08-18 at 1 13 17 PM" src="https://github.com/user-attachments/assets/1f631dc8-c392-44b0-9131-8213f64d9129">

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -24,6 +23,12 @@
+    font-family: Inter;
+    font-size: 16px;
+    font-weight: 600;
+    line-height: 19.36px;
```
px указываем ВСЕГДА целым числом. Округляем в большую или меньшую сторону. Здесь 19.36, значит ставим 19

shepherd1422 replied:
я сделал)
<img width="281" alt="Screenshot 2024-08-18 at 1 16 06 PM" src="https://github.com/user-attachments/assets/ee6ef9a6-a7ec-41bf-afb9-7bfc59097c12">

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -53,6 +64,14 @@
+    font-family: Inter;
+    font-size: 14px;
+    font-weight: 600;
+    line-height: 16.94px;
```
Идентичная ошибка с px, как и выше и так по всему проекту

shepherd1422 replied:
я сделал)
<img width="306" alt="Screenshot 2024-08-18 at 1 21 08 PM" src="https://github.com/user-attachments/assets/8cfff1c1-3e25-48aa-ab37-39089f492083">

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -67,9 +86,46 @@
-    margin: 0 auto    ;
+    margin: 0 auto ;
+}
+.benefits {
+    margin-top: 31px;
+    margin-bottom: 31px;
+    &__content-wrapper {
+        display: flex;
```
Между каждыми новыми классами (Здесь например между 98 и 99 строками) оставляем пустую строку для читаемости кода

shepherd1422 replied:
я сделал)
<img width="411" alt="Screenshot 2024-08-18 at 1 25 04 PM" src="https://github.com/user-attachments/assets/3486d590-9d66-4fa4-875f-79696bf344bd">

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -119,29 +194,127 @@
+    text-align: left;
+
+}
+.item-cards-2{
+    position: relative;
+}
+.item-cards-2-action{
+    position: absolute;
+    width: 100%;
+    display: flex;
+    justify-content: space-between;
+    margin-top: 247px;
+}
+.item-cards-2-action-left{
+    width: 60px;
+    height: 60px;
+    display: flex;
+    box-shadow: 0px 20px 50px 0px #3C3C3B12;
+
+
+}
+.item-cards-2-action-right{
+    width: 60px;
+    height: 60px;
+    display: flex;
+    box-shadow: 0px 20px 50px 0px #3C3C3B12;
+
+}
```
Стараемся не использовать цифры в названиях классов

pocahontassss replied:
Также здесь на 219 и 227 присутствует дублирование кода. Можно было просто навесить один и тот же класс на два тега, добавить модификаторы (--left и --right) для читаемости и все

shepherd1422 replied:
> Стараемся не использовать цифры в названиях классов

А можно исползовать цифры как текск ? например,,  вместо (item-cards-2) написать (items-cards__second-column) или (another__item-cards). ?

shepherd1422 replied:
> Стараемся не использовать цифры в названиях классов

![надо__подумать](https://github.com/user-attachments/assets/0cd93fb7-3cbd-4b70-b7f7-45ef23380b2b)

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -119,29 +194,127 @@
+.item-cards__grid-2{
+    display: flex;
+    width: 100%;
+    height: 429px;
+    max-height: 428px;
```
высота 429 пикселей и максимальная высота 428 пикселей. Как-то не сходится)

shepherd1422 replied:
Аха я не смог найти код (но буду иметь ввиду)

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -119,29 +194,127 @@
     height: 200px;
+    display: flex;
     padding: 15px;
     box-sizing: border-box;
```
Два раза указал высоту на 256 и 258 строках.
Так же зачем указывать max-width и max-height? Дублируется код

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -119,29 +194,127 @@
     padding: 15px;
     box-sizing: border-box;
+}
+.item-cards__grid-2-block{
+    display: flex;
+    flex-direction: column;
+    justify-content: end;
+    width: 309px;
+    height: 429px;
+    padding: 0px 0px 0px 0px;
+    gap: 13px;
+    box-shadow: 0px 20px 50px 0px #12112714;
+    background-image: url(/./assets/image13.png);
+    background-repeat: no-repeat ;
+    overflow: hidden;
+    background-position-x: center;
+    background-position-y: 35px;
+    padding: 20px;
+    box-sizing: border-box;
```
`background-image: url(/./assets/image13.png);` пишем в кавычках
То-есть правильный код выглядит так:
`background-image: url('/./assets/image13.png');`

pocahontassss replied:
Так же как и с body и font-family, box-sizing: border box используем для всех элементов на странице. Для этого в общие стили (style.scss) указываем такой код:
*,
*::before,
*::after {
  box-sizing: border-box;
}

Так же советую данные настройки:

`body {
  font-family: Inter, Arial, sans-serif; (Основной шрифт, используемый на странице)
  font-size: 14px; (Основной размер текста на странице)
  color: #000000; (Основной цвет текст на странице)
  margin: 0; 
  padding: 0;
}

a {
  text-decoration: none;
  color: inherit;
}`

body - для настройки общих стилей, что бы не прописывать их постоянно
a - убирает на теге <a> нижнее подчеркивание и синий цвет.

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -119,29 +194,127 @@
+    background-position-y: 35px;
+    padding: 20px;
+    box-sizing: border-box;
+}
+.item-cards__grid-2-block-element-1{
+    display: flex;
+    align-self: center;
+    padding-bottom: 131px;
+}
+.second-version{
+    display: flex;
+    align-self: center;
+    padding-bottom: 79px
+}
+.item-cards__grid-2-block-element-2{
+    display: flex;
+    font-family: Inter;
+    font-size: 20px;
+    font-weight: 600;
+
+}
+.item-cards__grid-2-block-element-3{
+    display: flex;
+    font-family: Inter;
+    font-size: 16px;
+    font-weight: 500;
+}
+.item-cards__grid-2-block-element-4{
+    display: flex;
+    width: 256px;
+    height: 73px;
+    background-image: url(/assets/Frame\ 39.png);
+    background-repeat: no-repeat ;
+    overflow: hidden;
+}
```
Между фигурной скобкой и названием класса ставим пробел для читаемости, то-есть:
`.item-cards__grid-2-block-element-4 {`
И опять же повторюсь, между классами ставим пустую строку (Например между 302 и 303, 296 и 297 и тд)

pocahontassss replied:
Так же на 308 строке лишний пробел после no-repeat

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -119,29 +194,127 @@
+    background-repeat: no-repeat ;
+    overflow: hidden;
 }
 .item-cards__grid-right-block-text{
```
Уже слишком странные названия пошли. Советую почитать про наименования классов. 
https://docs.google.com/document/d/1-QCBUsKZ0eQpbujyhkwUxL8E396mVOKRKhu1hvhgVTQ/edit
https://tpverstak.ru/common-css-class-names/

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -181,6 +354,9 @@
 background-position-x: right;
     background-position-y: bottom;
```
Поплыла табуляция

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -246,3 +422,178 @@ background-position-x: right;
     align-self: self-end;
     align-items: center;
 }
+@mixin clear-button{
+    box-shadow: none;
+    border: none;
+    background-color: transparent;
+    font-family: inherit;
+    padding: 0;
+    margin: 0;
+}
+@mixin clear-input {
+    font-family: inherit;
+    font-size: inherit;
+    background-color: transparent;
+    background-image: none;
+    box-shadow: none;
+    border: none;
+    border-radius: 0;
+    margin: 0;
+    padding: 0;
+    appearance: none;
+    outline: none;
+    }
```
Миксины нужно размещать в отдельном файле. Например создать файл normalize.scss и импортировать его в данный scss файл

pocahontassss commented on src/app/app.component.scss:
```diff
@@ -246,3 +422,178 @@ background-position-x: right;
+        &__contacts-content {
+            display: flex;
+            flex-direction: column;
+            gap: 8p x;
```
лишний пробел между px, в следствии чего свойство не сработало

pocahontassss commented on src/app/app.component.html:
```diff
@@ -52,10 +52,41 @@
+        <div class="benefits__content-wrapper">
+            <div class="benefits__content">
+                <div class="benefits__content-icon">
+                    <img src="../assets/icon-1.svg" alt="">
```
Если не хочешь использовать alt, тогда просто убирай данное свойство у тега

pocahontassss commented on src/app/app.component.html:
```diff
@@ -52,10 +52,41 @@
+            </div>
+            <div class="benefits__content">
+                <div class="benefits__content-icon">
+                    <img src="../assets/icon-3.svg" alt="">
```
Нужно правильно именовать картинки.
https://docs.google.com/document/u/0/d/1ipRxAtW0KEC5bj-vWLNk-p8AwjsN0AWmN8YApyP9VJw/mobilebasic
Здесь про наименование файлов (В конце статьи) и так же как правильно указывать путь к картинке

pocahontassss commented on src/app/app.component.html:
```diff
@@ -64,89 +95,352 @@
+
+                    <div class="item-cards__grid-right-block">
+                        <div class="item-cards__grid-right-block-4-text">
+                            <div class="item-cards__grid-right-block-text5">
+                                Розетки и выключатели
+                            </div>
```
Везде ты используешь div для размещения слов. Зачем? Для этого предназначены теги `<span>` и `<p>`. Это теги, предназначены для текстовых элементов. Так правильнее. div - пустая оболочка, инструмент для группировки нескольких элементов. Можно и для одного, например в div засунуть что-либо, но не делать из div - текстовый тег

pocahontassss commented on src/app/app.component.html:
```diff
@@ -64,89 +95,352 @@
-                        <div class="item-cards__grid-right-block-text5">
-                            Розетки и выключатели
+<section class="footer">
+    <div class="footer__container">
+        <div class="container">
```
Можно не создавать дополнительный div, а сразу использовать два класса для одного div'a. (Через пробел классы вставляем), Здесь нужно сделать так:
`class="footer__container container"`
Благодаря этому у одного div'a будут какие-то свои свойства и центровка (с помощью второго класса)

pocahontassss commented on src/app/app.component.html:
```diff
@@ -64,89 +95,352 @@
+                            <div class="footer__contacts-icon">
+                                <img src="../assets/call-calling.svg" alt="" class="footer__contacts-image">
+                            </div>
+                            <div class="footer__contacts-content">
+                                <span class="footer__contacts-title">Контактный телефон</span> 
+                                <span class="footer_contacts-phone">+ 7 (812) 309-09-34</span>
+                                <span class="footer_contacts-phone ">+ 7 (965) 084-29-09</span>
+                            </div>
```
Здесь моя ошибка. На лекции не обратил внимание на это.
Здесь нужно использовать для телефонов тег <a>, так как это ссылка, которая должна куда-то вести.
Здесь мы используем тег <a> со свойством href="tel:+781230090434"
то-есть итоговый код выглядит так:
`<a class="footer_contacts-phone" href="tel:+78123090434">+ 7 (812) 309-09-34</a>`
`<a class="footer_contacts-phone" href="tel:+79650842909">+ 7 (965) 084-29-09</a>`
  Благодаря этому по нажатию на данные ссылки пользователю вылезет предложение позвонить

pocahontassss commented on src/app/app.component.html:
```diff
@@ -64,89 +95,352 @@
+                                <img src="../assets/Icon.svg" alt="" class="footer__contacts-image">
+                            </div>
+                            <div class="footer__contacts-content">
+                                <span class="footer__contacts-title">Электронная почта</span> 
+                                <span class="footer_contacts-mail">tustin78.mail.ru</span>
+                                <span class="footer_contacts-mail">info.tustin.ru</span>
```
Аналогично выше, так же с href="mailto:"

pocahontassss commented on src/app/app.component.html:
```diff
@@ -64,89 +95,352 @@
+            <div class="footer__inline-horizontal"></div>
+            <div class="footer__navigation-wrapper">
+                <nav class="footer__navigation" >
+                    <span class="footer__navigation-item">Главная</span>
+                    <span class="footer__navigation-item">О Компании</span>
+                    <span class="footer__navigation-item">Каталог</span>
```
Также моя ошибка, здесь span должны быть внутри тега <a>, это тоже ссылки, которые ведут на другую страницу.

pocahontassss commented:
ИТОГО:

Пересмотреть мои комментарии и исправить ошибки.
Полностью пересмотреть нейминг файлов, классов. (Подумать, какие классы можно было бы переназвать)
Исправить форматирование, табуляцию.

shepherd1422 committed:
pocahontassss 2 days ago Стараемся не использовать цифры в названиях классов

пофиксил до этого места

shepherd1422 committed:
Улучшил (Форматирование)

🛠️ Прокачал верстку: теперь наш сайт выглядит как босс! 💪🔥

shepherd1422 committed:
Далше улучшил Форматирование Стилей

Сократил и Упростил Код

shepherd1422 committed:
Далше улучшил Форматирование Стилей

🚀 Верстка на высшем уровне: наш сайт теперь как ракета, готовая к взлёту! 🌌🛸

shepherd1422 committed:
Only (.item-cards__grid__wrap )—formation left😅

Почему домашние задания по программированию похожи на зомби-апокалипсис? Потому что они приходят волнами, пытаются съесть твой мозг, но настоящий герой-программист никогда не сдаётся и всегда находит способ выжить! 🧟‍♂️💻

shepherd1422 committed:
changed all incompatible names and image names

Why did the student bring a ladder to school?

Because they heard the teacher say, “Never quit, always aim high!” 😄

shepherd1422 committed:
убрал абстрактно наименованные классы до минимума. проектировал общее форматирование для внутри-корпораативного применения(меньше БЕМ - болше стилей сортированных  в общий блок).

😅

shepherd1422 committed:
div were organized - span were added when necessary

shepherd1422 committed:
Решил ( Comment on lines 428 to 430)

😅

shepherd1422 commented:
Я не смог убрать имя Alt возле фотографии img

shepherd1422 commented:
Я не смог убрать имя Alt возле фотографии img

shepherd1422 committed:
Revert "Решил ( Comment on lines 428 to 430)"

This reverts commit 8a2221c6faa3a40966782b7f0d599d4af231b1b0.

Co-Authored-By: pocahontassss <162623615+pocahontassss@users.noreply.github.com>

shepherd1422 committed:
Renamed most of content blocks to ul-li-span for better semantics

it is because brother Vladislav recommended to give more attention to correct sementics of the web page for good SEO optimization. But it may be less importand inside enterprize projects

Co-Authored-By: pocahontassss <162623615+pocahontassss@users.noreply.github.com>

shepherd1422 committed:
still hammering gitbash push origin
